### PR TITLE
Add RGB LED for Dozen0

### PR DESCRIPTION
**Pull request by yynmt**
## Description

It is a change for the specification change of the new PCB.
The new PCB now supports RGB LED.
Along with this, the default keymap has also been changed.
These changes maintain backward compatibility with older PCB.

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [x] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/dozen0/keymaps/default/keymap.c [hunk 1]
```diff
@@ -15,20 +15,110 @@
...
+   */
+  [_ADJUST] = LAYOUT(
+    RGB_TOG, RGB_MOD, RGB_HUI, RGB_SAI, RGB_VAI, _______, \
+    RGBRST, RGB_RMOD, RGB_HUD, RGB_SAD, RGB_VAD, _______ \
+  )
 };
 
-bool process_record_user(uint16_t keycode, keyrecord_t *record) {
-  return true;
+uint32_t layer_state_set_user(uint32_t state) {
```
Comment (noroadsleft) [inline keyboards/dozen0/keymaps/default/keymap.c]: `layer_state_set_user` is typedef'd now. (Reduces firmware size if there are only a few layers.)

```suggestion
layer_state_t layer_state_set_user(layer_state_t state) {
```

File: keyboards/dozen0/rules.mk [hunk 2]
```diff
@@ -1,30 +1,77 @@
...
+#     source code.
+#
+#     If no clock division is performed on the input clock inside the AVR (via the
+#     CPU clock adjust registers or the clock division fuses), this will be equal to F_CPU.
+F_USB = $(F_CPU)
+
+# Interrupt driven control endpoint task(+60)
+OPT_DEFS += -DINTERRUPT_CONTROL_ENDPOINT
+
+
```
Comment (noroadsleft) [inline keyboards/dozen0/rules.mk]: Lines 4-40 can all be deleted. QMK will set these up for you based on `MCU = atmega32u4`.

https://github.com/qmk/qmk_firmware/blob/9ea9806d67dd3dfd775cf854b9becd35f3cdc748/quantum/mcu_selection.mk#L39-L75

File: keyboards/dozen0/rules.mk [hunk 3]
```diff
@@ -1,30 +1,77 @@
...
 # Bootloader selection
 #   Teensy       halfkay
 #   Pro Micro    caterina
 #   Atmel DFU    atmel-dfu
 #   LUFA DFU     lufa-dfu
 #   QMK DFU      qmk-dfu
-#   ATmega32A    bootloadHID
-#   ATmega328P   USBasp
-BOOTLOADER = atmel-dfu
+#   atmega32a    bootloadHID
```
Comment (noroadsleft) [inline keyboards/dozen0/rules.mk]:
```suggestion
#   ATmega32A    bootloadHID
#   ATmega328P   USBasp
```

File: keyboards/dozen0/rules.mk [hunk 4]
```diff
@@ -1,30 +1,77 @@
...
+
+# If you don't know the bootloader type, then you can specify the
+# Boot Section Size in *bytes* by uncommenting out the OPT_DEFS line
+#   Teensy halfKay      512
+#   Teensy++ halfKay    1024
+#   Atmel DFU loader    4096
+#   LUFA bootloader     4096
+#   USBaspLoader        2048
+# OPT_DEFS += -DBOOTLOADER_SIZE=4096
+
```
Comment (noroadsleft) [inline keyboards/dozen0/rules.mk]: Lines 50-60 can all be deleted.

File: keyboards/dozen0/rules.mk [hunk 5]
```diff
@@ -1,30 +1,77 @@
...
 # Do not enable SLEEP_LED_ENABLE. it uses the same timer as BACKLIGHT_ENABLE
 SLEEP_LED_ENABLE = no       # Breathing sleep LED during USB suspend
 # if this doesn't work, see here: https://github.com/tmk/tmk_keyboard/wiki/FAQ#nkro-doesnt-work
 NKRO_ENABLE = no            # USB Nkey Rollover
 BACKLIGHT_ENABLE = no       # Enable keyboard backlight functionality on B7 by default
-RGBLIGHT_ENABLE = no        # Enable keyboard RGB underglow
+RGBLIGHT_ENABLE = yes        # Enable keyboard RGB underglow
 MIDI_ENABLE = no            # MIDI support (+2400 to 4200, depending on config)
 UNICODE_ENABLE = no         # Unicode
 BLUETOOTH_ENABLE = no       # Enable Bluetooth with the Adafruit EZ-Key HID
```
Comment (noroadsleft) [inline keyboards/dozen0/rules.mk]: Please change Lines 64-77 to:

```Makefile
BOOTMAGIC_ENABLE = no       # Virtual DIP switch configuration
MOUSEKEY_ENABLE = yes       # Mouse keys
EXTRAKEY_ENABLE = yes       # Audio control and System control
CONSOLE_ENABLE = no         # Console for debug
COMMAND_ENABLE = no         # Commands for debug and configuration
# Do not enable SLEEP_LED_ENABLE. it uses the same timer as BACKLIGHT_ENABLE
SLEEP_LED_ENABLE = no       # Breathing sleep LED during USB suspend
# if this doesn't work, see here: https://github.com/tmk/tmk_keyboard/wiki/FAQ#nkro-doesnt-work
NKRO_ENABLE = no            # USB Nkey Rollover
BACKLIGHT_ENABLE = no       # Enable keyboard backlight functionality
RGBLIGHT_ENABLE = yes       # Enable keyboard RGB underglow
MIDI_ENABLE = no            # MIDI support
UNICODE_ENABLE = no         # Unicode
BLUETOOTH_ENABLE = no       # Enable Bluetooth with the Adafruit EZ-Key HID
AUDIO_ENABLE = no           # Audio output on port C6
FAUXCLICKY_ENABLE = no      # Use buzzer to emulate clicky switches
HD44780_ENABLE = no         # Enable support for HD44780 based LCDs
```

This removes the firmware size impact numbers (they're rarely accurate) and visually aligns the comments.

**Comment by yynmt** [comment]
Thank you for the review.
I fixed them.

File: keyboards/dozen0/rules.mk [hunk 6]
```diff
@@ -8,26 +8,25 @@ MCU = atmega32u4
 #   LUFA DFU     lufa-dfu
 #   QMK DFU      qmk-dfu
 #   ATmega32A    bootloadHID
-#   ATmega328P   USBasp
-BOOTLOADER = atmel-dfu
+BOOTLOADER = caterina
```
Comment (noroadsleft) [inline keyboards/dozen0/rules.mk]: Lines 10-11 should be changed to:

```Makefile
#   ATmega32A    bootloadHID
#   ATmega328P   USBasp
BOOTLOADER = caterina
```

There was a line addition here on the previous review that didn't get added.

Comment (yynmt) [reply keyboards/dozen0/rules.mk]: I'm sorry, I overlooked that.

File: keyboards/dozen0/keymaps/default/keymap.c [hunk 7]
```diff
@@ -15,20 +15,110 @@
...
+   * |  Cut | Copy |Paste |  Up  |Delete| Bksp |
+   * |      |      |      |      |      |Raise |
+   * |------+------+------+------+------+------|
+   * | Ctrl |Shift | Left | Down |Right |Enter |
+   * |      |      |      |      |      |Lower |
+   * `-----------------------------------------'
+   */
+	[_BASE] = LAYOUT(
+    LCTL(KC_X), LCTL(KC_C), LCTL(KC_V), KC_UP, KC_DEL, LT(_RAISE,KC_BSPC), \
+    KC_LCTL, KC_LSFT, KC_LEFT, KC_DOWN, KC_RGHT, LT(_LOWER,KC_ENT) \
```
Comment (fauxpark) [inline keyboards/dozen0/keymaps/default/keymap.c]: The backslashes at the end of these lines are not necessary. You can keep them if you want as it isn't hurting anything, but IMO it looks neater without them.

**Comment by fauxpark** [comment]
Thanks!

**Comment by yynmt** [comment]
Thank you very much everyone.

**Pull request merged**